controllers/dspipeline_controller.go

```diff
@@ -197,57 +200,203 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
-	usingCustomDB := params.UsingExternalDB(dspa)
+	// Initialize conditions
+	conditions := []metav1.Condition{}
 
+	usingCustomDB := params.UsingExternalDB(dspa)
+	databaseReady := r.buildCondition("DatabaseReady", dspa.Generation, "DatabaseReady")
 	if !usingCustomDB {
+
 		err = r.ReconcileDatabase(ctx, dspa, params)
 		if err != nil {
+			databaseReady.Status = metav1.ConditionFalse
+			databaseReady.Reason = "DataBaseFailedToDeploy"
 			return ctrl.Result{}, err
 		}
+
+		if !r.isDeploymentAvailable(ctx, dspa, "mariadb") {
+			databaseReady.Status = metav1.ConditionFalse
+			databaseReady.Reason = "DataBaseFailedToDeploy"
+
+			err = r.Status().Update(ctx, dspa)
+			if err != nil {
+				log.Info(err.Error())
+				return ctrl.Result{}, err
+			}
+		}
+
+		conditions = append(conditions, databaseReady)
 	}
 
 	usingCustomStorage := params.UsingExternalStorage(dspa)
-
+	objectStorageReady := r.buildCondition("ObjectStorageReady", dspa.Generation, "ObjectStorageReady")
 	if !usingCustomStorage {
 		err := r.ReconcileStorage(ctx, dspa, params)
 		if err != nil {
+			objectStorageReady.Status = metav1.ConditionFalse
+			objectStorageReady.Reason = "ObjectStorageFailedToDeploy"
 			return ctrl.Result{}, err
 		}
+
+		if !r.isDeploymentAvailable(ctx, dspa, "minio") {
+			objectStorageReady.Status = metav1.ConditionFalse
+			objectStorageReady.Reason = "ObjectStorageFailedToDeploy"
+
+			err = r.Status().Update(ctx, dspa)
```

Comment:
is it a good idea to update the condition several times in each reconcile loop? can we do it all at once (unless there is an error being reported and we are exiting reconcile loop anyways), each update will queue a new reconcile loop, might as well do it all at once and only queue it once, no?